From 625810d806e734ec35dfa503a4c6bfa03f6ff569 Mon Sep 17 00:00:00 2001
From: Paolo Bonzini <pbonzini@redhat.com>
Date: Tue, 22 Jul 2014 14:37:12 -0400
Subject: [virt] kvm: add missing cleanup_srcu_struct

Message-id: <1406040016-3289-38-git-send-email-pbonzini@redhat.com>
Patchwork-id: 86116
O-Subject: [RHEL7 PATCH v2 037/221] KVM: add missing cleanup_srcu_struct
Bugzilla: 1116936
RH-Acked-by: Radim Krcmar <rkrcmar@redhat.com>
RH-Acked-by: Bandan Das <bsd@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Reported-by: hrg <hrgstephen@gmail.com>
Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
(cherry picked from commit 820b3fcdeb80d30410f4427d2cbf9161c35fdeef)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/virt/kvm/kvm_main.c b/virt/kvm/kvm_main.c
index 8da4269..f3cc4df 100644
--- a/virt/kvm/kvm_main.c
+++ b/virt/kvm/kvm_main.c
@@ -606,6 +606,7 @@ static void kvm_destroy_vm(struct kvm *kvm)
  kvm_arch_destroy_vm(kvm);
  kvm_destroy_devices(kvm);
  kvm_free_physmem(kvm);
+ cleanup_srcu_struct(&kvm->irq_srcu);
  cleanup_srcu_struct(&kvm->srcu);
  kvm_arch_free_vm(kvm);
  hardware_disable_all();
-- 
1.7.1